darwin/dataset/upload_manager.py

```diff
@@ -186,6 +210,82 @@ def full_path(self) -> str:
         return construct_full_path(self.data["path"], self.data["filename"])
 
 
+class MultiFileItem:
+    def __init__(
+        self, directory: Path, files: List[Path], merge_mode: ItemMergeMode, fps: int
+    ):
+        self.directory = directory
+        self.name = directory.name
+        self.files = [LocalFile(file, fps=fps) for file in files]
+        self.merge_mode = merge_mode
+        self.layout = self._create_layout()
+
+    def _create_layout(self):
+        """
+        Creates the layout to be used when uploading the files as a dataset item:
+        - For multi-slotted items: LayoutV2
+        - For series items: LayoutV2, but only with `.dcm` files
+        - For multi-channel items: LayoutV3
+
+        Raises
+        ------
+        ValueError
+            - If no DICOM files are found in the directory for `ItemMergeMode.SERIES` items
+            - If the number of files is greater than 16 for `ItemMergeMode.CHANNELS` items
+        """
+        if self.merge_mode == ItemMergeMode.SLOTS:
+            num_files = len(self.files)
+            layout_shape = LAYOUT_SHAPE_MAP.get(num_files, [4, 4])
+            return {
+                "version": 2,
+                "slots": [str(i) for i in range(num_files)],
+                "type": "grid" if num_files >= 4 else "horizontal",
+                "layout_shape": layout_shape,
+            }
+        elif self.merge_mode == ItemMergeMode.SERIES:
+            self.files = [
+                file for file in self.files if file.local_path.suffix.lower() == ".dcm"
+            ]
+            if not self.files:
+                raise ValueError("No `.dcm` files found in 1st level of directory")
+            return {
+                "version": 2,
+                "slots": [self.name],
+                "type": "grid",
+            }
+        elif self.merge_mode == ItemMergeMode.CHANNELS:
+            if len(self.files) > 16:
+                raise ValueError(
+                    f"No multi-channel item can have more than 16 files. The following directory has {len(self.files)} files: {self.directory}"
+                )
+            return {
+                "version": 3,
+                "slots_grid": [[[file.local_path.name for file in self.files]]],
+            }
+
+    # Need to add a `data` attribute for MultFileItem? Where do we get `fps` from?
+    def serialize_v2(self):
+        optional_properties = ["fps"]
```

Comment:
`optional_properties` only contains 1 item for now, but this is to mirror how the `serialize_v2()` method works for `LocalFile`. If we add support for other such as `as_frames` or `extract_views` in future, will be easy to extend